### PR TITLE
Makes RemoveComponent more sane

### PR DESCRIPTION
**Pull request by DGamerL**
## What Does This PR Do
Our current way to properly remove a component from something is to `qdel` it. This does work fully as intended, but it doesn't make a lot of sense.
To make it a bit more logical I renamed `RemoveComponent` to `UnlinkComponent`. It "unlinks" the component from the parent so it can be reassigned. This seemed more logical to me.

`RemoveComponent` has been changed into a wrapper for `qdel(src)`. Same functionality, but now it looks more logical.

## Why It's Good For The Game
It makes more sense now

## Testing
To test some various ones:
- Xeno eggs 
- Admin/normal outfits

None of these runtimed when tested

<hr>

### Declaration

- [x] I confirm that I either do not require [pre-approval](https://github.com/ParadiseSS13/Paradise/blob/master/docs/CODE_OF_CONDUCT.md#types-of-changes-that-need-approval) for this PR, or I have obtained such approval and have included a screenshot to demonstrate this below.
  
  
  <hr>

## Changelog

NPFC

### CODE REVIEW

**Review by lewcc — APPROVED** [review]

File: code/datums/components/surgery_initiator.dm [hunk 1]
```diff
@@ -68,7 +68,6 @@
 	var/obj/item/P = parent
 	if(!P.sharp)
 		RemoveComponent()
```
[the remark this answers is not in the document]
Comment (DGamerL) [reply code/datums/components/surgery_initiator.dm:70]: Is this needed? IIRC deleting the component also calls the unlinking procs. It'd also mean that all other places that use `RemoveComponent` should unlink first unless GC issues, but they don't

[the remark this answers is not in the document]
Comment (Contrabang) [reply code/datums/components/surgery_initiator.dm:70]: I think its best for parity, just in case

File: code/datums/components/_component.dm [hunk 2]
```diff
@@ -441,6 +441,14 @@
 		return new_comp
 	return old_comp
 
+/**
+  * Removes the component from the datum
+  */
+/datum/proc/DeleteComponent(component_to_nuke)
+	var/datum/component/removing = GetComponent(component_to_nuke)
+	if(istype(removing))
```
Comment (Contrabang) [inline code/datums/components/_component.dm:449]:
```suggestion
	if(istype(removing, component_to_nuke)
```

File: code/datums/components/_component.dm [hunk 3]
```diff
@@ -441,6 +441,14 @@
 		return new_comp
 	return old_comp
 
+/**
+  * Removes the component from the datum
+  */
+/datum/proc/DeleteComponent(component_to_nuke)
+	var/datum/component/removing = GetComponent(component_to_nuke)
+	if(istype(removing, component_to_nuke)
```
Comment (DGamerL) [inline code/datums/components/_component.dm:449]:
```suggestion
	if(istype(removing, component_to_nuke))
```

File: code/datums/components/_component.dm [hunk 4]
```diff
@@ -441,6 +441,14 @@
 		return new_comp
 	return old_comp
 
+/**
+  * Removes the component from the datum
+  */
+/datum/proc/DeleteComponent(component_to_nuke)
+	var/datum/component/removing = GetComponent(component_to_nuke)
+	if(istype(removing, component_to_nuke))
```
Comment (Contrabang) [inline code/datums/components/_component.dm:449]:
```suggestion
	if(istype(removing, component_to_nuke) && !QDELETED(removing))
```

File: code/datums/components/_component.dm [hunk 5]
```diff
@@ -467,6 +475,12 @@
 	parent = null
 	SEND_SIGNAL(old_parent, COMSIG_COMPONENT_REMOVING, src)
 
+/**
+  * Deletes the component and removes it from parent.
+  */
+/datum/component/proc/RemoveComponent() // This really is just a wrapper to pretend that we're using sane procs to fully remove a component
+	qdel(src)
```
Comment (Contrabang) [inline code/datums/components/_component.dm:482]:
```suggestion
	if(!QDELETED(src))
		qdel(src)
```

**Review by Contrabang — CHANGES_REQUESTED** [review]
Last things

**Review by Contrabang — APPROVED** [review]

**Review by lewcc — APPROVED** [review]

**Pull request merged**